runtime/coverage.go

```diff
@@ -235,8 +235,8 @@ func (r *CoverageReport) MarshalJSON() ([]byte, error) {
 	coverage := make(map[string]LC, len(r.Coverage))
 	for location, locationCoverage := range r.Coverage { // nolint:maprange
 		typeID := location.TypeID(nil, "")
-		locationID := typeID[:len(typeID)-1]
-		coverage[string(locationID)] = LC{
+		locationID := fmt.Sprintf("%v%v", typeID[:2], location.String())
```

Comment:
It took me a while to understand what was needed here and how it was generated.

Basically, we want a string representation of the location, including the location prefix, so it can be differentiated. However, `Location` `String()` implementations do not include the prefix.

Instead of preparing this location ID here, it might be better to just introduce a new method for `Location`, e.g. `ID()`. This also addresses the issue that a location prefix might be longer than one character.

Also, could you please add some tests? Given that the format changes here, I was expecting some tests to have to be adjusted as well.

Reply:
> Also, could you please add some tests? Given that the format changes here, I was expecting some tests to have to be adjusted as well.

Will do. I was also surprised to see the tests passing without any change. I guess because on emulator we are dealing with `AddressLocation`, and on tests with `StringLocation` & `ScriptLocation`. I will check it though :+1:

Reply:
I have updated all the `common.Location` implementations, to include an `ID` method. I have also added tests both in the coverage package, and in each specific implementation of the `common.Location` interface :pray:

Reply:
Nice!